src/plugins/irc-events/link.js

```diff
@@ -170,16 +170,24 @@ function emitPreview(client, msg, preview) {
 	});
 }
 
-function fetch(uri, cb) {
+function getRequestHeaders(client) {
+	const headers = {
+		"User-Agent": "Mozilla/5.0 (compatible; The Lounge IRC Client; +https://github.com/thelounge/lounge)",
+	};
+	if (client.language !== null) {
+		headers["Accept-Language"] = client.language;
+	}
+	return headers;
+}
+
+function fetch(uri, client, cb) {
```

Comment:
`client` is only passed to get the language, so it seems like a heavy thing to do (I'd like to avoid passing things like `client`, `clientManager`, etc. too much, it makes code even more coupled than it currently is).

`function fetch(uri, language, cb)` would do, or we could go ES6 (it's gonna be for v3 anyway, where we drop Node v4), and do `function fetch (uri, {language}, cb)`, that you call with `fetch(preview.link, {language: client.language}, function(res) {` (I know `fetch(preview.link, client, function(res) {` would still work, but it's less explicit and still passes the whole thing around).

Reply:
fixed, went with functon parameter destructuring